src/routes/docs/products/databases/bulk-operations/+page.markdoc

```diff
@@ -0,0 +1,150 @@
+
+{% info title="Permissions required" %}
+You must grant **update** permissions to users at the **collection level** before users can update documents.
+[Learn more about permissions](/docs/products/databases/permissions)
+{% /info %}
+
+You can update multiple documents in a single request using the `updateDocuments` method.
+
+{% multicode %}
+```server-nodejs
+const sdk = require('node-appwrite');
+
+const client = new sdk.Client()
+    .setEndpoint('https://<REGION>.cloud.appwrite.io/v1')
+    .setProject('<PROJECT_ID>')
+    .setKey('<API_KEY>');
+
+const databases = new sdk.Databases(client);
+
+const result = await databases.updateDocuments(
+    '<DATABASE_ID>',
+    '[COLLECTION_ID]',
+    [
+        {
+            documentId: 'document-id-1',
+            data: { name: 'Updated Document 1' }
+        },
+        {
+            documentId: 'document-id-2',
+            data: { name: 'Updated Document 2' }
+        }
+    ]
+);
+```
+{% /multicode %}
+
+# Upsert documents {% #upsert-documents %}
+
+{% info title="Permissions required" %}
+You must grant **create** and **update** permissions to users at the **collection level** before users can create documents.
+[Learn more about permissions](/docs/products/databases/permissions)
+{% /info %}
+
+You can upsert multiple documents in a single request using the `upsertDocuments` method.
+
+{% multicode %}
+```server-nodejs
+const sdk = require('node-appwrite');
+
+const client = new sdk.Client()
+    .setEndpoint('https://<REGION>.cloud.appwrite.io/v1')
+    .setProject('<PROJECT_ID>')
+    .setKey('<API_KEY>');
+
+const databases = new sdk.Databases(client);
+
+const result = await databases.upsertDocuments(
+    '<DATABASE_ID>',
+    '[COLLECTION_ID]',
+    [
+        {
+            documentId: sdk.ID.unique(),
+            data: { name: 'New Document 1' }
+        },
+        {
+            documentId: 'document-id-2', // Existing document ID
+            data: { name: 'New Document 2' }
+        }
+    ]
+);
+```
+{% /multicode %}
+
+# Delete documents {% #delete-documents %}
+
+{% info title="Permissions required" %}
+You must grant **delete** permissions to users at the **collection level** before users can delete documents.
+[Learn more about permissions](/docs/products/databases/permissions)
+{% /info %}
+
+You can delete multiple documents in a single request using the `deleteDocuments` method.
+
+{% multicode %}
+```server-nodejs
+const sdk = require('node-appwrite');
+
+const client = new sdk.Client()
+    .setEndpoint('https://<REGION>.cloud.appwrite.io/v1')
+    .setProject('<PROJECT_ID>')
+    .setKey('<API_KEY>');
+
+const databases = new sdk.Databases(client);
+
+const result = await databases.deleteDocuments(
+    '<DATABASE_ID>',
+    '[COLLECTION_ID]',
+    [] // Queries
+);
+```
+{% /multicode %}
+
+**Note:** When deleting documents, you can specify queries to filter which documents to delete. If no queries are provided, all documents in the collection will be deleted.
```

Comment:
I am thinking this should be highlighted a bit more aggressively because I do expect some cases where a user just misses this and proceeds to delete everything 😕

Reply:
I'll convert this to an info note then

That component should stand out more